### PR TITLE
feat(gas): Updates to support Ruby 3.3

### PR DESCRIPTION
**Pull request by dazuma**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -15,6 +15,8 @@ jobs:
         include:
           - os: ubuntu-latest
             ruby: "3.1"
+          - os: ubuntu-latest
+            ruby: "3.3"
```
Comment (chadlwilson) [inline .github/workflows/ci.yml:19]: Might be a chicken vs. egg problem here based on the failure from GHA - `/opt/hostedtoolcache/Ruby/3.3.0/x64/lib/ruby/3.3.0/bundled_gems.rb:74:in `require': libruby.so.3.1: cannot open shared object file: No such file or directory - /tmp/d20240110-3852-1dyayf/vendor/gems/google-protobuf-3.25.2-x86_64-linux/lib/google/protobuf_c.so (LoadError)` given that version doesn't declare compatibility with `3.3`?

Perhaps can only test with Ruby 3.2 for now?

Comment (dazuma) [reply .github/workflows/ci.yml:19]: Ah, thanks @chadlwilson, I'd forgotten about that. I suppose that _could_ explain why these are failing, although I'm not yet convinced. The test tries to bypass the fact that the require is coming from a gem, adding the lib directly to the load path directly. So the gemspec metadata shouldn't factor in. And it does look like the shared object is being built and does exist here, so I'm not yet sure why it won't load. I'll poke at this some more tomorrow. Really want to put an actual 3.3 test in place if we can.

Comment (chadlwilson) [reply .github/workflows/ci.yml:19]: Yeah, you're certainly more of an expert than me, but for whatever reason, when compiling the extension it seems to be doing `/usr/local/rvm/rubies/ruby-3.1.3/bin/ruby -I. ../../../../../ext/google/protobuf_c/extconf.rb` so presumably linking against `3.1`. Not sure if that's a missing `rvm use` or runner or environment problem or what.

Perhaps something like that is always why `3.2` wasn't being specifically tested earlier either?

Agree it'd be great to sanity test these against multiple rubies.

Comment (dazuma) [reply .github/workflows/ci.yml:19]: Thanks for noting that. It was in fact linking against 3.1; I was able to verify it loads the shared object under 3.1 but fails under 3.3. But the odd thing is, if I set `RUBY_CC_VERSION` to multiple versions, it does link properly against the requested versions, including 3.3, but if it is set to a single version, that version is always linked against 3.1 (which is, I think, the native Ruby version in the docker image), regardless of the actual value of `RUBY_CC_VERSION`. There still seems to be some deeper issue here, possibly a bug in rake-compiler-dock. For now, I updated the test to cross-compile against multiple Ruby versions to work around the issue (and for our use cases, we always compile against multiple versions so it should be fine), but I'll open an issue to investigate further.

**Pull request merged**